src/server/scripts/Events/hallows_end.cpp

```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
         }
     };
 
-    AuraScript* GetAuraScript() const
+    AuraScript* GetAuraScript() const override
     {
         return new spell_hallows_end_has_water_bucket_AuraScript();
     }
 };
 
+class spell_hallows_end_bucket_lands : public SpellScriptLoader
+{
+public:
+    spell_hallows_end_bucket_lands() : SpellScriptLoader("spell_hallows_end_bucket_lands") {}
+
+    class spell_hallows_end_bucket_lands_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_hallows_end_bucket_lands_SpellScript);
+
+        bool handled;
+        bool Load() { handled = false; return true; }
```

Comment:
So that the same bucket cast doesn't give buckets of water to more than one player hit by the spell.